lib/multi-acc-new-eks-mixed-observability-pattern/pipeline.ts

```diff
@@ -0,0 +1,388 @@
+    logger.debug(`Retrieved CDK context ${JSON.stringify(cdkContext)}`);
+    return cdkContext;
+}
+
+/*
+* prodEnv1: PROD1 account ID and region
+* prodEnv2: PROD2 account ID and region 
+* monitoringEnv: MON account ID and region 
+ */
+export interface PipelineMultiEnvMonitoringProps {
+    prodEnv1: cdk.Environment;
+    prodEnv2: cdk.Environment;
+    pipelineEnv: cdk.Environment;
+    monitoringEnv: cdk.Environment;
+}
+
+/*
+ * Main multi-account monitoring pipeline.
+ */
+export class PipelineMultiEnvMonitoring {
+
+    readonly pipelineRegion = process.env.COA_PIPELINE_REGION! || process.env.CDK_DEFAULT_REGION!;
+    
+    async buildAsync(scope: Construct) {
+
+        // All Grafana Dashboard URLs from `cdk.json` if present
+        clusterDashUrl = utils.valueFromContext(scope, "cluster.dashboard.url", undefined);
+        kubeletDashUrl = utils.valueFromContext(scope, "kubelet.dashboard.url", undefined);
+        namespaceWorkloadsDashUrl = utils.valueFromContext(scope, "namespaceworkloads.dashboard.url", undefined);
+        nodeExporterDashUrl = utils.valueFromContext(scope, "nodeexporter.dashboard.url", undefined);
+        nodesDashUrl = utils.valueFromContext(scope, "nodes.dashboard.url", undefined);
+        workloadsDashUrl = utils.valueFromContext(scope, "workloads.dashboard.url", undefined);        
+
+        // Checks for Git Owner
+        if (!this.pipelineRegion) {
+            logger.debug("ERROR: COA_PIPELINE_REGION or CDK_DEFAULT_REGION environment variable is not defined.");
+            process.exit(101);
+        }
+
+        // environments IDs consts
+        const context = await populateAccountWithContextDefaults(this.pipelineRegion);
+
+        ampAccount = context.prodEnv1.account as string;
+        ampRegion = context.prodEnv1.region as string;
+        cwAccount = context.prodEnv2.account as string;
+        cwRegion = context.prodEnv2.region as string;
+        monAccount = context.monitoringEnv.account as string;
+        monRegion = context.monitoringEnv.region as string;
+
+        const PROD1_ENV_ID = `coa-eks-prod1-${context.prodEnv1.region}`;
+        const PROD2_ENV_ID = `coa-eks-prod2-${context.prodEnv2.region}`;
+        const MON_ENV_ID = `coa-cntrl-mon-${context.monitoringEnv.region}`;
+        
+        // Get AMG info from SSM SecureString
+        const amgInfo = JSON.parse(await getSSMSecureString('/cdk-accelerator/amg-info',this.pipelineRegion))['amg'];
+        amgWorkspaceUrl = amgInfo.workspaceURL;
+        const amgWorkspaceIAMRoleARN = amgInfo.workspaceIAMRoleARN;
+
+        // Props for cross-account trust role in PROD1 account to trust AMG from MON account, inorder to access PROD1's AMP
+        ampAssumeRoleName = "AMPAccessForTrustedAMGRole";
+        const AMPAccessRoleStackProps: CreateIAMRoleNestedStackProps = {
+            roleName: ampAssumeRoleName!,
+            trustArn: amgWorkspaceIAMRoleARN!,
+            actions: [
+                "aps:ListWorkspaces",
```

Comment:
Please move this as a file ourside and render it. Check bluperints addons for examples

Reply:
Here I am using create-iam-role.ts (a generic nestedstack class to create IAM role with required name, trustARN, actions and resources) to create 2 cross-account access roles. trustARN here is AMG's IAM role. Please correct me if I didn't understand this comment properly.

Reply:
done - moved outside